### PR TITLE
feat(did): support BLS-12-381 verification methods

### PR DESCRIPTION
**Pull request by gsora**
This PR adds support for BLS-12-381 public keys in DID verification methods.

The CLI `tx did add-verification-method` now excepts a `--verification-method-type` parameter with the type of `VerificationMethod` to be used for parsing/storing on chain.

Types accepted:

- `Bls12381G1Key2020`
- `Bls12381G2Key2020`
- `EcdsaSecp256k1VerificationKey2019` (previous default, compatible with a Cosmos public key)

### CODE REVIEW

File: x/did/client/cli/tx.go [hunk 1]
```diff
@@ -114,51 +118,166 @@ func NewCreateDidDocumentCmd() *cobra.Command {
...
-				pk.Address().Bytes(),
-			))
+			didID := did.NewChainDID(clientCtx.ChainID, id)
+
+			blsVerificationBuilder := func(pkBytes []byte) (*did.Verification, error) {
+				pkHash := sha256.Sum256(pkBytes)
+				var pkHashStr string
+				pkHashStr, err = multibase.Encode(multibase.Base58BTC, pkHash[:])
+				if err != nil {
+					return nil, fmt.Errorf("cannot encode pubkey hash to string, %w", err)
```
Comment (barriebyron) [inline x/did/client/cli/tx.go:175]: @gsora  not sure what or how this error message shows up, but I like seeing real language help users solve problems. If "cannot encode pubkey hash to string" then what can a user do to make the encoding work? 

Curious mind here 

(and by the way, I love error messages and think they have very high content value)

File: x/did/client/cli/tx.go [hunk 2]
```diff
@@ -114,51 +118,166 @@ func NewCreateDidDocumentCmd() *cobra.Command {
...
+						verificationType,
+					),
+					[]string{did.Authentication},
+					nil,
+				), nil
+			}
+
+			switch verificationType {
+			case did.CosmosAccountAddress, did.Ed25519VerificationKey2018, did.X25519KeyAgreementKey2019:
+				return fmt.Errorf("verification type type %s not supported", verificationType)
```
Comment (barriebyron) [inline x/did/client/cli/tx.go:195]: me again, your friendly error message reviewer! 
If a specific verification type is not supported, can we share the supported types and a helpful hint to use a supported verification type?

**Review by barriebyron — COMMENTED** [review]
Hi team! did you know that error messages are near and dear to my heart, as they are to every user? 

I asked a few questions for error messages I spied in this PR.

To help our users, error messages might go something like this:

[The error] [How to fix it]

I'm always up for collaborating on helping our users :)

File: x/did/client/cli/tx.go (hunk 1 above)
Comment (gsora) [reply x/did/client/cli/tx.go:175]: > then what can a user do to make the encoding work?

Good question! Usually this error pops out when the user is trying to add a BLS key via the command line.

Supposedly the user also created this key via some sort of CLI command, external to the node... 

If the key isn't well-formed, there's not much the user can do except maybe recreating it.

File: x/did/client/cli/tx.go (hunk 2 above)
Comment (gsora) [reply x/did/client/cli/tx.go:195]: We already do! If you try using a unsupported verification key, this message pops up: 

`Error: invalid or missing verification type, allowed types: Bls12381G1Key2020, Bls12381G2Key2020, EcdsaSecp256k1VerificationKey2019`

File: go.mod [hunk 3]
```diff
@@ -4,12 +4,14 @@ go 1.18
 
 require (
 	github.com/AlecAivazis/survey/v2 v2.3.4
+	github.com/coinbase/kryptology v1.8.0
```
Comment (PaddyMc) [inline go.mod:7]: This PR is ace, but I think we need to consider making a separate repo for SSI client-related stuff such as this. :thinking: 

Should we be adding this dependency to the node if it's only for a client?

**Pull request closed without merging**